tests/Unit/Utilities/Test_ContainerHelpers.cpp

```diff
@@ -3,34 +3,87 @@
 
 #include "tests/Unit/TestingFramework.hpp"
 
+#include <complex>
 #include <cstddef>
 #include <vector>
 
 #include "DataStructures/DataVector.hpp"
 #include "Utilities/ContainerHelpers.hpp"
 
+namespace {
+struct HalfDataVectorSize {
+  size_t operator()(const DataVector& x) noexcept { return x.size() / 2; }
+};
+
+struct ReverseIndexStlVector {
+  template <typename T>
+  decltype(auto) operator()(T& x, size_t i) noexcept {
+    return x.at(x.size() - i - 1);
+  }
+};
+
+}  // namespace
 SPECTRE_TEST_CASE("Unit.Utilities.ContainerHelpers", "[Unit][Utilities]") {
-  const std::vector<double> a{0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0};
-  const double b = 10.0;
-  const DataVector c(10, 2.0);
-  std::vector<double> a2{0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0};
-  double b2 = 10.0;
-  DataVector c2(10, 2.0);
-  REQUIRE(get_size(a) == 10);
-  REQUIRE(get_size(b) == 1);
-  REQUIRE(get_size(c) == 10);
-  REQUIRE(get_size(a2) == 10);
-  REQUIRE(get_size(b2) == 1);
-  REQUIRE(get_size(c2) == 10);
-  for (size_t i = 0; i < a.size(); ++i) {
-    get_element(a2, i) = get_element(a, i) * 2.0;
-    get_element(b2, i) = get_element(b, i) * 2.0;
-    get_element(c2, i) = get_element(c, i) * 2.0;
-    CHECK(get_element(a, i) == static_cast<double>(i));
-    CHECK(get_element(b, i) == 10.0);
-    CHECK(get_element(c, i) == 2.0);
-    CHECK(get_element(a2, i) == 2.0 * static_cast<double>(i));
-    CHECK(get_element(b2, i) == 20.0);
-    CHECK(get_element(c2, i) == 4.0);
+  const std::vector<double> constant_stl_vector{0.0, 1.0, 2.0, 3.0, 4.0,
+                                                5.0, 6.0, 7.0, 8.0, 9.0};
+  const double constant_fundamental = 10.0;
+  const std::complex<double> constant_complex_of_fundamental =
+      std::complex<double>(1.0, 5.0);
+  const DataVector constant_spectre_vector(10, 2.0);
+  std::vector<double> stl_vector{0.0, 1.0, 2.0, 3.0, 4.0,
+                                 5.0, 6.0, 7.0, 8.0, 9.0};
+  double fundamental = 10.0;
+  std::complex<double> complex_of_fundamental = std::complex<double>(1.0, 5.0);
+  DataVector spectre_vector(10, 2.0);
+  CHECK(get_size(constant_stl_vector) == 10);
```

Comment:
Wouldn't it be preferable to continue the checks in case the `get_size` merely erroneously returns 1 but the container is properly initialized with several elements?